File: Remove-Element/Remove-Element.py


class Solution:
    # @param    A       a list of integers
    # @param    elem    an integer, value need to be removed
    # @return an integer
    def removeElement(self, A, elem):
        n=len(A)
        i=0
        while i!=n:
            if A[i]==elem:
                A[i],A[n-1]=A[n-1],A[i]
                i=i-1
                n=n-1
            i=i+1
        return n
        
 

File: solutions/Remove-Element/Remove-Element.cpp

class Solution {
public:
    int removeElement(int A[], int n, int elem) {
        int ans=0;
        for(int i=0;i<n;i++)
            if(A[i]!=elem)
                A[ans++]=A[i];
        return ans;
    }
};


 